sentry/src/main/java/io/sentry/ConnectionFactory.java

```diff
@@ -0,0 +1,26 @@
+package io.sentry;
+
+import io.sentry.transport.Connection;
+import org.jetbrains.annotations.NotNull;
+
+public final class ConnectionFactory {
+  private ConnectionFactory() {}
+
+  public static @NotNull Connection create(final @NotNull SentryOptions options) {
```

Comment:
This is one way to determine which client should be used, the other would be to make it more explicit for users.